File: backend/app/models/assistant.py
import enum
from datetime import UTC, datetime
from uuid import uuid4

from sqlalchemy import Boolean, DateTime, Enum, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class CareReminderCategory(str, enum.Enum):
    WATER = "WATER"
    PAUSE = "PAUSE"
    BREATHING = "BREATHING"
    REST = "REST"
    ROUTINE = "ROUTINE"
    CUSTOM = "CUSTOM"


class CareReminder(Base):
    __tablename__ = "bergmann_care_reminders"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid4()))
    user_id: Mapped[str] = mapped_column(ForeignKey("bergmann_users.id", ondelete="CASCADE"), index=True, nullable=False)
    title: Mapped[str] = mapped_column(String(120), nullable=False)
    category: Mapped[CareReminderCategory] = mapped_column(Enum(CareReminderCategory), nullable=False)
    cadence: Mapped[str] = mapped_column(String(32), default="DAILY", nullable=False)
    time_local: Mapped[str | None] = mapped_column(String(5), nullable=True)
    note: Mapped[str | None] = mapped_column(Text, nullable=True)
    active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    last_completed_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(UTC), nullable=False)
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        default=lambda: datetime.now(UTC),
        onupdate=lambda: datetime.now(UTC),
        nullable=False,
    )

    user = relationship("User", back_populates="care_reminders")

File: frontend/src/lib/assistant.ts
import { apiRequest } from "@/lib/api";

export type CareReminderCategory = "WATER" | "PAUSE" | "BREATHING" | "REST" | "ROUTINE" | "CUSTOM";

export type CareReminder = {
  id: string;
  title: string;
  category: CareReminderCategory;
  cadence: string;
  time_local: string | null;
  note: string | null;
  active: boolean;
  last_completed_at: string | null;
  created_at: string;
  updated_at: string;
};

export async function createCareReminder(input: {
  title: string;
  category: CareReminderCategory;
  cadence?: string;
  time_local?: string | null;
  note?: string | null;
}): Promise<CareReminder> {
  return apiRequest<CareReminder>("/assistant/reminders", {
    method: "POST",
    body: JSON.stringify({
      title: input.title,
      category: input.category,
      cadence: input.cadence ?? "DAILY",
      time_local: input.time_local ?? null,
      note: input.note ?? null
    })
  });
}

export async function listCareReminders(): Promise<CareReminder[]> {
  return apiRequest<CareReminder[]>("/assistant/reminders");
}

export async function completeCareReminder(reminderId: string): Promise<CareReminder> {
  return apiRequest<CareReminder>(`/assistant/reminders/${reminderId}/complete`, {
    method: "POST"
  });
}
